### PR TITLE
Multiple fixes to Auto-Type for 2.5.1

### LINKED ISSUE #3688: Hide "Sequence" column from appearing in "Auto-Type" popup if not needed
## Summary
[NOTE]: # ( Provide a brief overview of what the new feature is all about )
As in the title, an option to disable the "Sequence" column from appearing in the "Auto-Type" popup when multiple entries exist.

## Desired Behavior
[NOTE]: # ( Tell us how the new feature should work, be specific )
Hide "Sequence" column in "Auto-Type" popup.

## Context
[NOTE]: # ( Why does this feature matter to you? What unique circumstances do you have? )
It's awfully distracting to see the Auto-Type sequence when trying to pick an entry from the list.
I know this sort of option was/is available in KeePass for Windows, but I've forgotten how it's done.
I can no longer check to see as I have no machines running Windows and run Linux full-time.

A picture of the column in question:
DisableSequenceColumnKeePassXC

### LINKED ISSUE #3689: KeePassXC wants to record the screen.
I updated KeePassXC to 2.5.0 with Homebrew on macOS:

~~~
==> Upgrading 1 outdated package:
keepassxc 2.4.3 -> 2.5.0
==> Upgrading keepassxc
==> Satisfying dependencies
==> Downloading https://github.com/keepassxreboot/keepassxc/releases/download/2.5.0/KeePassXC-2.5.0.d
==> Downloading from https://github-production-release-asset-2e65be.s3.amazonaws.com/52729242/7624cb0
######################################################################## 100.0%
==> Verifying SHA-256 checksum for Cask 'keepassxc'.
==> Backing App 'KeePassXC.app' up to '/usr/local/Caskroom/keepassxc/2.4.3/KeePassXC.app'.
==> Removing App '/Applications/KeePassXC.app'.
==> Unlinking Binary '/usr/local/bin/keepassxc-cli'.
~~~

And when I run it, Catalina shows this warning:

Screen Shot 2019-10-27 at 17 38 52

Why?

### PR DESCRIPTION
**Pull request by droidmonkey**
[TIP]:  # ( Provide a general summary of your changes in the title above ^^ )

## Type of change
[NOTE]: # ( Please remove all lines which don't apply. )
- ✅ Bug fix (non-breaking change which fixes an issue)
- ✅ Refactor (significant modification to existing code)

## Description and Context
[NOTE]: # ( Describe your changes in detail, why is this change required? )
[NOTE]: # ( Describe the context of your change. Explain large code modifications. )
[NOTE]: # ( If it fixes an open issue, please add "Fixes #XXX" as necessary )
* Hide Auto-Type sequences column when unnecessary
  * Fix #3688 - hide the sequences column if all of the entry matches return the same sequence. This cleans up redundent data in the Auto-Type selection dialog introduced in 2.5.0.

* Revert "Remove Carbon from Mac Auto-Type (#3347)"
  * Removing Carbon API was premature. The NSEvent API was causing shortcut capture issues due to it being a key monitor not an interceptor.

* macOS: Check for Auto-Type permissions on use instead of at launch
  * Fix #3689 - link the use of Auto-Type with the permissions required to use it

NOTE: you can reset the screen recorder permissions using the following command: `tccutil reset ScreenCapture`

## Screenshots
[TIP]:  # ( Do not include screenshots of your actual database! )
Dialog for entry-level Auto-Type:
image

Dialog for global Auto-Type:
image

## Testing strategy
[NOTE]: # ( Please describe in detail how you tested your changes. )
[TIP]:  # ( We expect new code to be covered by unit tests and documented with doc blocks! )
Tested on macOS 10.15

## Checklist:
[NOTE]: # ( Please go over all the following points. )
[NOTE]: # ( Again, remove any lines which don't apply. )
[NOTE]: # ( Pull Requests that don't fulfill all [REQUIRED] requisites are likely )
[NOTE]: # ( to be sent back to you for correction or will be rejected.  )
- ✅ I have read the **CONTRIBUTING** document. **[REQUIRED]**
- ✅ My code follows the code style of this project. **[REQUIRED]**
- ✅ All new and existing tests passed. **[REQUIRED]**
- ✅ I have compiled and verified my code with `-DWITH_ASAN=ON`. **[REQUIRED]**
- ✅ My change requires a change to the documentation, and I have updated it accordingly.
- ✅ I have added tests to cover my changes.

### CODE REVIEW

File: src/autotype/mac/AutoTypeMac.cpp [hunk 1]
```diff
@@ -18,13 +18,18 @@
...
+#include "gui/MessageBox.h"
 
 #include <ApplicationServices/ApplicationServices.h>
 
 #define HOTKEY_ID 1
 #define MAX_WINDOW_TITLE_LENGTH 1024
 #define INVALID_KEYCODE 0xFFFF
 
+namespace {
+    bool accessibilityChecked = false;
```
Comment (phoerious) [inline src/autotype/mac/AutoTypeMac.cpp]: Do not indent namespaces.

File: src/autotype/mac/AutoTypeMac.cpp [hunk 2]
```diff
@@ -470,6 +479,21 @@ OSStatus AutoTypePlatformMac::hotkeyHandler(EventHandlerCallRef nextHandler, Eve
 {
     Q_UNUSED(nextHandler);
 
+    // Determine if the user has given proper permissions to KeePassXC to perform Auto-Type
+    if (!accessibilityChecked) {
+        if (macUtils()->enableAccessibility() && macUtils()->enableScreenRecording()) {
```
Comment (phoerious) [inline src/autotype/mac/AutoTypeMac.cpp]: So despite reverting to Carbon, we still need these permissions, right?

Comment (droidmonkey) [reply src/autotype/mac/AutoTypeMac.cpp]: Yah carbon had nothing to do with the actual typing (accessibility) and window title matching (screen recording)

File: src/autotype/mac/AutoTypeMac.cpp (hunk 1 above)
Comment (droidmonkey) [reply src/autotype/mac/AutoTypeMac.cpp]: fixed

**Pull request merged**